gulpfile.js

```diff
@@ -0,0 +1,305 @@
+var gulp                = require('gulp');
+var browserSync         = require('browser-sync');
+var del                 = require('del');
+var vinylPaths          = require('vinyl-paths');
+var typescript          = require('gulp-typescript');
+var concat              = require('gulp-concat');
+var source              = require('vinyl-source-stream');
+var buffer              = require('vinyl-buffer');
+var wiredep             = require('wiredep').stream;
+var inject              = require('gulp-inject');
+var nodemon             = require('gulp-nodemon');
+var jshint              = require('gulp-jshint');
+var jscs                = require('gulp-jscs');
+var complexity          = require('gulp-complexity');
+var karma               = require('karma').server;
+var stylish             = require('jshint-stylish');
+var eslint              = require('gulp-eslint');
+var eslintPathFormatter = require('eslint-path-formatter');
+var mocha               = require('gulp-mocha');
+var  mochaLcovReporter  = require('mocha-lcov-reporter');
+var coverage            = require('gulp-coverage');
+var open                = require('gulp-open');
+var istanbul            = require('gulp-istanbul');
+// var Promise          = require('Bluebird');
+var merge               = require('gulp-merge');
+
+var CONFIG              = require('./build.config');
+
+gulp.task('hello', function()
+{
+	console.log('Waaazzuuuuuppp');
+});
+
+// **********************************************************************
+// **********************************************************************
+// **********************************************************************
+
+// tells you everything wrong with your code, quickly. Format, styling, and complexity.
+gulp.task('analyze', function() {
+  return gulp.src(CONFIG.client.sourceFiles)
+    .pipe(jshint())
+    .pipe(jshint.reporter('jshint-stylish'))
+    .pipe(jshint.reporter('fail'))
+    .on('error', function(e)
+    {
```

Comment:
this is the last one :)
We have to agree explicitly of style guide, that determines also such subtle things like starting { in the same line with command, or on the next line. There are more establishing styles, or we can agree on our own. I would prefer to pick one existing.